src/keybind.cpp

```diff
@@ -1635,19 +1625,19 @@ void	kf_JumpToResourceExtractor()
 	/* not supported if a spectator */
 	SPECTATOR_NO_OP();
 
-	if (psOldRE && (STRUCTURE *)psOldRE->psNextFunc)
+	if (psOldRE.has_value() && *psOldRE != apsExtractorLists[selectedPlayer].end())
 	{
-		psOldRE = psOldRE->psNextFunc;
+		++(*psOldRE);
 	}
 	else
 	{
-		psOldRE = apsExtractorLists[selectedPlayer];
+		psOldRE.emplace(apsExtractorLists[selectedPlayer].begin());
 	}
 
-	if (psOldRE)
+	if (*psOldRE != apsExtractorLists[selectedPlayer].end() && **psOldRE != nullptr)
```

Comment:
I'm hitting the addConsoleMessage() below this every time it reaches the end of the extractor list (tested on Beta 1). Attempting another jump will succeed.

Reply:
Thanks for spotting this. I'll take a look.

Reply:
Fixed.